client/src/app/shared/components/projector/projector.component.ts

```diff
@@ -177,7 +175,20 @@ export class ProjectorComponent extends BaseComponent implements OnDestroy {
         this._projector = projector;
         if (projector?.id !== this.oldProjectorId) {
             this.oldProjectorId = projector?.id;
-            this.updateProjectionSubscription();
+            this.slides = this.projector.current_projections_as_observable.pipe(
+                map(projectionList => {
+                    return projectionList.map(projection => {
+                        return {
+                            collection: collectionFromFqid(projection.content_object_id),
+                            data: projection.content,
+                            stable: !!projection.stable,
+                            type: projection.type || '',
+                            options: projection.options || {},
+                            ...(!!projection.content?.error && { error: projection.content.error })
+                        } as SlideData;
+                    });
+                })
```

Comment:
I replaced `updateSlides()`, returns an observable now, reduces the logic quite a bit

Reply:
Nice, I will  try to use `switchMap` to chain a new projector-observable with the `current_projections_as_observable`.